lib/params/VolumeIsoParams.cpp

```diff
@@ -22,17 +22,7 @@ VolumeIsoParams::VolumeIsoParams(
 
 VolumeIsoParams::VolumeIsoParams(
 	DataMgr *dataMgr, ParamsBase::StateSave *ssave, XmlNode *node
-) : VolumeParams(dataMgr, ssave, node) {
-	SetDiagMsg("VolumeIsoParams::VolumeIsoParams() this=%p", this);
-
-	// If node isn't tagged correctly we correct the tag and reinitialize
-	// from scratch;
-	//
-	if (node->GetTag() != VolumeIsoParams::GetClassType()) {
```

Comment:
if this branch is evaluating to true something deeper is probably wrong. What is the the value node->GetTab()?

Reply:
The problem is that this “failsafe” code that exists in some Params classes breaks if you use inheritance and VolumeIsoParams inherits from VolumeParams. It calls the parent constructor which says I’m not a “VolumeParams“ so let’s delete everything and reinitialize as VolumeParams. Then VolumeIsoParams says I’m not a “VolumeIsoParams“ so let’s delete everything and reinitialize as VolumeIsoParams.

Reply:
The problem is that this “failsafe” code that exists in some Params classes breaks if you use inheritance and VolumeIsoParams inherits from VolumeParams. It calls the parent constructor which says I’m not a “VolumeParams“ so let’s delete everything and reinitialize as VolumeParams. Then VolumeIsoParams says I’m not a “VolumeIsoParams“ so let’s delete everything and reinitialize as VolumeIsoParams.

Reply:
The failsafe code exists for backwards compatibility with old session files. If a new renderer is added to vapor and a user tries to use a session file generated from a previous version what will the behavior be? I don't think the application will work as expected. Is there another way to work around this problem while preserving backwards compatibility with older session files?

Reply:
In theory, yes I see what you mean, but first, we don't have backwards compatibility with old session files, and second, this code would not improve the situation in any way: This code only checks that a node's tag is the same as the class name, however the node would never be created from a session file if it does not have the same tag, Vapor would just assume it is a different renderer.

Reply:
However if you prefer, I can keep it and work around it.

Reply:
OK. Agreed. Let's remove the failsafe from all of the code since it's liable to problematic elsewhere in the future, shall we?

Reply:
Sounds good. Done. (Fix #2040)